### DISCUSSION TITLE
CSV_XS not thread safe

### DISCUSSION THREAD

**Discussion by Tux** [tomk3003/spreadsheet-compare#2] [General]
https://github.com/tomk3003/spreadsheet-compare/blob/317cba5be34ebb16f5325fc06abc8dcec62af8bf/script/spreadcomp#L109

Not an issue yet in Text::CSV_XS?
Do you have example code so I can try to make it thread-safe?

Sorry to hijack an issue to ask

**Comment by tomk3003** [comment]
I just use threading under Windows with Mojo::IOLoop::Thread and had failing tests that went away when using Text::CSV::PP.
Then I read [RT #119312](https://rt.cpan.org/Public/Bug/Display.html?id=119312) and decided to wait for a fix for that.

I'll try to come up with a small example showing the problem. But it will take a day or two.

**Comment by tomk3003** [comment]
converting to discussion

**Comment by tomk3003** [comment]
Running the csv tests of the module I can trigger an error every 10 runs or so, but I was not able to get an error with an isolated example. 
But I found, that threading under windows is so slow, that it isn't worth the effort anyway. A standard loop is much faster.

I think I'll just disable parallel processing under Windows and remove the warning.
